bofire/data_models/features/categorical.py

```diff
@@ -404,9 +419,14 @@ def validate_experimental(self, values: pd.Series) -> pd.Series:
             )
         return values
 
-    def to_dict(self) -> Dict:
-        """Returns the catergories and corresponding objective values as dictionary"""
-        return dict(zip(self.categories, self.objective))
-
     def __call__(self, values: pd.Series) -> pd.Series:
-        return values.map(self.to_dict()).astype(float)
+        if self.objective is None:
+            return pd.Series(
+                data=[np.nan for _ in range(len(values))],
```

Comment:
we need a test for this method.

Reply:
It will be in features/test_categorical.py